plugins/history_plugin/include/eosio/history_plugin/history_plugin.hpp

```diff
@@ -18,6 +18,25 @@ namespace eosio {
 
 namespace history_apis {
 
+class read_write {
+   history_const_ptr history;
+
+   public:
+      read_write(history_const_ptr&& history)
+        : history(history) {}
+
+      struct prune_history_params {
+        int32_t             height;     // height of the last block we want to prune data of
+      };
+
+      struct prune_history_result {
+        int64_t             last_id;   // last id pruned
```

Comment:
Since `prune` doesn't take an `id` this is not helpful. How about just returning the # of records removed.